includes/init_includes/init_customer_auth.php

```diff
@@ -79,27 +79,34 @@
  * 2 = customer authorization pending can browse but no prices
  */
 switch (true) {
+  /**
+   * bypass redirects for these scripts, to processing regardless of store mode or cust auth mode
+   */
+  case (preg_match('/ipn_main_handler\.php$/', $_SERVER['SCRIPT_NAME'])):

```

Comment:
I suppose changing the delimiter to `|` could be good, and then also make the beginning `/` become a literal character, thus attempting to ensure that it's a match on the complete filename and not just a suffix to a filename.

Or should it be even more generic, as in `|_handler\.php$|` so that other scripts can be easily supported?

I had also considered that it might be ideal to add notifier hook support so that additional plugins could register themselves for inclusion here. Haven't figured out how best to do that yet, so was leaving that for a separate PR.